### PR TITLE
support (instance) IAM roles

### PR DESCRIPTION
**Pull request by Mortinke**
#4 Static IAM (user) credentials are no longer necessary. In addition, the static credentials are no longer a characteristic that you want to push RDS logs.

### CODE REVIEW

File: scripts/go.bash [hunk 1]
```diff
@@ -21,7 +21,7 @@ if [[ -z $LOGZIO_TOKEN ]]; then
     exit 1
 fi
 
-if ! [[ -z $AWS_ACCESS_KEY && -z $AWS_SECRET_KEY ]]; then
+if ! [[ -z $RDS_LOG_FILE && -z $RDS_SLOW_LOG_FILE && -z $RDS_ERROR_LOG_FILE ]]; then
```
Comment (ofer-velich) [inline scripts/go.bash]: no need for that

File: scripts/base.sh [hunk 2]
```diff
@@ -88,16 +88,18 @@ function parse_arguments {
...
 function configure_rds() {
     mkdir -p /root/.aws
     echo "[default]" > $AWS_CREDENTIAL_FILE
     echo "region = $AWS_REGION" >> $AWS_CREDENTIAL_FILE
-    echo "aws_secret_access_key = $AWS_SECRET_KEY" >> $AWS_CREDENTIAL_FILE
-    echo "aws_access_key_id = $AWS_ACCESS_KEY" >> $AWS_CREDENTIAL_FILE
+    if [[ ! -z $AWS_SECRET_KEY ]]; then
+      echo "aws_secret_access_key = $AWS_SECRET_KEY" >> $AWS_CREDENTIAL_FILE
+      echo "aws_access_key_id = $AWS_ACCESS_KEY" >> $AWS_CREDENTIAL_FILE
+    fi
```
Comment (ofer-velich) [inline scripts/base.sh]: please add a log if we are depending on IAM role

File: scripts/go.bash (hunk 1 above)
Comment (Mortinke) [reply scripts/go.bash]: If we want to use IAM roles, no values are passed for _AWS_ACCESS_KEY_ & _AWS_SECRET_KEY_. Thus, this condition would be true and configure_rds would never be called. Therefore, another characteristic is required to indicate "I want to ship RDS logs".

File: scripts/base.sh (hunk 2 above)
Comment (Mortinke) [reply scripts/base.sh]: done

File: scripts/go.bash (hunk 1 above)
Comment (ofer-velich) [reply scripts/go.bash]: why do you need any condition, i think you can just remove this condition, and let the configure_rds to be called. 
RDS_LOG_FILE, RDS_SLOW_LOG_FILE, RDS_ERROR_LOG_FILE will be validated on the sync_rds call, am i missing some thing ?

**Comment by mirii1994** [comment]
Closing, as this was handled in #8

**Pull request closed without merging**